xbmc/guilib/GUIFontTTF.cpp

```diff
@@ -732,7 +732,7 @@ bool CGUIFontTTFBase::CacheCharacter(wchar_t letter, uint32_t style, Character *
       m_posX += -bitGlyph->left;
 
     // check we have enough room for the character
-    if ((m_posX + bitGlyph->left + bitmap.width) > static_cast<int>(m_textureWidth))
+    if ((m_posX + bitGlyph->left + bitmap.width) > m_textureWidth)
```

Comment:
and the result being unsigned is the correct interpretation of c(++) rules. if you add signed and unsigned numbers, the result is unsigned. and bitmap.width is unsigned. is it signed in your freetype (ftimage.h, struct FT_Bitmap_) ?